### PR TITLE
Improve ux

### PR DESCRIPTION
**Pull request by rstammer**
A picture says more than words here. Wdyt, @carpodaster ? I followed your hint on overriding the `toString` method on `Board` and quite like it. Do you agree?

bildschirmfoto 2019-01-05 um 15 40 19

### CODE REVIEW

File: scala/cz-tictactoe.scala [hunk 1]
```diff
@@ -17,7 +17,7 @@ class Game() {
     // FIXME: make sure both players have different `marker` values
     val players = List[Player](player1, player2)
 
-    board.printField
+    print(board.toString)
```
Comment (carpodaster) [inline scala/cz-tictactoe.scala]: is `toString` maybe implicit like in Ruby? If yes, the explicit call could be omitted.

File: scala/cz-tictactoe.scala [hunk 2]
```diff
@@ -46,11 +46,26 @@ class Game() {
...
 
   def readMove(): Move = {
+    var x : Int = -1
+    var y : Int = -1
+
     print("Enter X coord: ")
-    val x = scala.io.StdIn.readInt() - 1
+    x = scala.io.StdIn.readInt() - 1
+
+    while (!(0 to 2 contains x)) {
```
Comment (carpodaster) [inline scala/cz-tictactoe.scala:55]: Uuh, nice trick :)

File: scala/cz-tictactoe.scala [hunk 3]
```diff
@@ -83,15 +98,21 @@ class Board() {
...
+
     for(i <- 0 until 3) {
+      board += ((i+1) + "    ")
       for(j <- 0 until 3) {
-        print(" " + fields(j)(i) + " ")
+        board += (" " + fields(j)(i) + " ")
       }
-      println
-      println
+      board += "\n\n"
```
Comment (carpodaster) [inline scala/cz-tictactoe.scala:113]: Would probably remove the two CR/LFs and let that be handled by the caller

**Review by carpodaster — APPROVED** [review]
Very nice!

File: scala/cz-tictactoe.scala (hunk 1 above)
Comment (rstammer) [reply scala/cz-tictactoe.scala]: Good idea, let's try!

File: scala/cz-tictactoe.scala (hunk 3 above)
Comment (rstammer) [reply scala/cz-tictactoe.scala:113]: Hm, that does not work, we need the newline breaks here!

**Pull request merged**